### DISCUSSION TITLE
How to prohibit parallel visits (overlapping visits) when delivering split demands?

### DISCUSSION THREAD

**Discussion by FernandoDeAlmeidaAc** [google/or-tools#3950] [Routing (and legacy CP) questions]
Hi everyone.

**Problem description:**
I have a VRP where some demands are greater than the vehicles' capacity. 

**What I tried:**
I split the big demands in multiples nodes to deliver them separately. However, these nodes that represent the same location are receiving visits from multiple vehicles at the same time, which the problem does not allow.

To avoid parallel visits, I implemented the following constraint:

```cpp
std::vector<IntervalVar*> intervals;
// Node 0 is the depot.
for (size_t node_index = 1; node_index < data.num_nodes; node_index++) {
    int64_t index = manager.NodeToIndex(RoutingIndexManager::NodeIndex(node_index));
    intervals.push_back(solver->MakeFixedDurationIntervalVar(
        time_dimension.CumulVar(index), data.service_time, "service_interval"));
}
solver->AddConstraint(solver->MakeDisjunctiveConstraint(
    intervals, "split_node_invervals"));
```

**What I cannot solve:**
I used the "MakeDisjunctiveConstraint" function to prohibit overlapping visits, but it doesn't work. I never used this function before, so maybe I'm doing something wrong. Is there a way to use this function differently or use another function to avoid parallel visits?

I'm implementing in c++.

**Comment by edders51** [comment]
Did you find a solution to this? I am trying to solve the same problem...

**Reply by FernandoDeAlmeidaAc** [reply]
I impose a minimal distance for each pair of divided nodes. It's not ideal if you break a node in many parts, but it works.

```
            auto diff = _solver->MakeDifference(
                _time_dimension->CumulVar(second_node), _time_dimension->CumulVar(first_node));

            _solver->AddConstraint(_solver->MakeGreaterOrEqual(diff, time_to_avoid_parallel_visit));
```

**Comment by Mizux** [comment]
You should model a ressource constraint: https://developers.google.com/optimization/routing/cvrptw_resources
